test/tools/unified-spec-runner/unified-utils.ts

```diff
@@ -110,17 +111,22 @@ export async function topologySatisfies(
   }
 
   if (typeof r.csfle === 'boolean') {
-    if (r.csfle) {
-      ok &&= config.clientSideEncryption.enabled;
+    const versionSupportsCSFLE = semverGte(config.version, '4.2.0');
+    const csfleEnabled = config.clientSideEncryption.enabled;
 
+    if (r.csfle) {
+      ok &&= versionSupportsCSFLE && csfleEnabled;
       if (!ok && skipReason == null) {
-        skipReason = `requires csfle to run but CSFLE is not set for this environment`;
+        skipReason = versionSupportsCSFLE
+          ? `requires csfle to run but CSFLE is not set for this environment`
+          : 'requires mongodb >= 4.2 to run csfle tests';
```

Comment:
Another fix to this logic - here's the actual spec https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#runonrequirement

The server must be >=4.2.0 AND csfle must be enabled for the driver.  For us, we rely on the configuration's `enabled` property to be consistent through all our tests.